pandas/util/_validators.py

```diff
@@ -320,3 +320,39 @@ def validate_axis_style_args(data, args, kwargs, arg_name, method_name):
         msg = "Cannot specify all of '{}', 'index', 'columns'."
         raise TypeError(msg.format(arg_name))
     return out
+
+
+def validate_fillna_kwargs(value, method, validate_scalar_dict_value=True):
+    """Validate the keyword arguments to 'fillna'.
+
+    This checks that exactly one of 'value' and 'method' is specified.
+    If 'method' is specified, this validates that it's a valid method.
+
+    Parameters
+    ----------
+    value, method : object
+        The 'value' and 'method' keyword arguments for 'fillna'.
+    validate_scalar_dict_value : bool, default True
+        Whether to validate that 'value' is a scalar or dict; specifically
+        that it is not a list or tuple.
+
+    Returns
+    -------
+    value, method : object
+    """
+    from pandas.core.missing import clean_fill_method
+
+    if value is None and method is None:
+        raise ValueError("Must specify a fill 'value' or 'method'.")
+    elif value is None and method is not None:
+        method = clean_fill_method(method)
+
+    elif value is not None and method is None:
+        if validate_scalar_dict_value and isinstance(value, (list, tuple)):
```

Comment:
This check doesn't seem especially useful / correct given the error message.

There are many things that are not scalars or dicts, but also not `lists` or `tuples`, so we would fail to hit this TypeError.

But I was trying to keep this as a straight refactor for NDFrame.fillna, with the only behavior change being `Categorical.fillna`.

Reply:
I dont' think this is odd to want to validate that a scalar can be only a certain type. maybe would change this 

``validate_scalar=lambda x: is_scalar(x) or is_dict_like(x)``

Reply:
this parameter name is terrible, ideally would like to change this

Reply:
Two things:

1. Don't call a change by a contributor "terrible". It's not helpful.
2. The error message says "'value' parameter must be a scalar or dict". That seems to match the parameter name pretty well.

Reply:
the parameter name is terrible, obviously NOT the contributor, which I didn't say or imply at all. It needs to be changed.

Reply:
The parameter name is certainly descriptive in what it does, *and* it is for an internal method, so it's not much of a problem it is that long. Why bother for the rest?